### ISSUE TITLE
Null exception for parameter "Path" in UWP

### ISSUE THREAD

**Issue by NikolaNel** [NateRickard/Plugin.AudioRecorder#8]
Using the plugin in UWP (Windows 10 for desktop and a Windows 10 mobile device)
I get the following exception when calling await recorder.StartRecording();

"{System.ArgumentNullException: Path cannot be null.Parameter name: path   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)   at Plugin.AudioRecorder.WaveRecorder.<StartRecorder>d__6.MoveNext()--- End of stack trace from previous location where exception was thrown ---   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)   at Plugin.AudioRecorder.AudioRecorderService.<StartRecording>d__43.MoveNext()--- End of stack trace from previous location where exception was thrown ---   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()   at atWork.ViewModels.HomeViewModel.<Record>d__40.MoveNext()}"

I have granted microphone and storage permissions, but the devices dont have external SD card.

**Comment by TristanMoreau** [comment]
In the AudioRecorderService constructor, the Init() method is called to get the path of the file where the audio will be saved. Problem is this Init() partial method is async on the UWP part. When you use StartRecording(), since the Init() isn't awaited, it can cause crashes if the path is still null.

If you're willing to recompile the library, you can use this quick (and dirty) fix to make it work by using a loop checking the state of the path in StartRecording() : 

```
public async Task<Task<string>> StartRecording ()
{
    //quickfix
    int retry = 0;
    while (filePath == null)
    {
          if (retry++ > 10)
               break;
          Task.Delay(100).Wait();
    }

    ResetAudioDetection();

    InitializeStream (PreferredSampleRate);

    await recorder.StartRecorder (audioStream, filePath);

    AudioStreamDetails = new AudioStreamDetails
    {
		ChannelCount = audioStream.ChannelCount,
		SampleRate = audioStream.SampleRate,
		BitsPerSample = audioStream.BitsPerSample
    };

    startTime = DateTime.Now;
    recordTask = new TaskCompletionSource<string> ();

    Debug.WriteLine ("AudioRecorderService.StartRecording() complete.  Audio is being recorded.");

    return recordTask.Task;
} 
```

You can of course adjust it to your needs as a temporary solution but, ideally, the Init() method should be awaited.

**Comment by KodeKooker** [comment]
I also found this blocking issue while using the package on UWP platform, I'm waiting for an official fix on the nuget package.

**Comment by NateRickard** [comment]
Should be fixed in #15, though I'm unable to release a package at the moment as my Windows machine needs to be rebuilt. If you need this fix, just build the project and use the DLLs directly until I have time to revisit.
